maxim/dataset/dataset.py

```
@@ -109,3 +116,171 @@ def validate_data_structure(
+    """
+
+    def __init__(self, api: "MaximAPI"):
+        """
+        Initialize Dataset with a MaximAPI instance.
+
+        Args:
+            api (MaximAPI): The MaximAPI instance to use for API calls
+        """
+        self.api = api
+
+    def _map_attachment_entries_to_entry_ids(
```

Ruff (0.12.2)

`_map_attachment_entries_to_entry_ids` is too complex (11 > 10)

(C901)